libsrc/lgd3d/d6Render.cpp

```diff
@@ -1,53 +1,275 @@
+#include <lgassert.h>
+#include <dbg.h>
 #include "d6Render.h"
+#include "d6States.h"
+#include "d6Prim.h"
+#include "d6Over.h"
+
+extern cD6Renderer*			pcRenderer;
+extern IDirect3DViewport3*	g_lpViewport;
+extern DWORD				g_dwScreenWidth;
+extern DWORD				g_dwScreenHeight;
+extern float				g_XOffset;
+extern float				g_YOffset;
+
+#define _BYTE  uint8
+#define BYTEn(x, n)   (*((_BYTE*)&(x)+n))
+#define BYTE1(x)   BYTEn(x,  1)         // byte 1 (counting from 0)
 
 cD6Renderer::cD6Renderer(DWORD dwInitialBufferType, DWORD* pdwRequestedFlags)
 {
+	m_bOverlaysOn = TRUE;
+
+	if (dwInitialBufferType)
+	{
+		if (dwInitialBufferType == 1)
+		{
+			pcRenderBuffer = cMSBuffer::Instance();
+			pcStates = cMSStates::Instance();
+		}
+	}
+	else
+	{
+		pcRenderBuffer = cImBuffer::Instance();
+		pcStates = cImStates::Instance();
+	}
+
+	DWORD RenderStatesSize = pcStates->GetRenderStatesSize();
+	CreateStatesStack(10, RenderStatesSize);
+	if (!pcStates->Initialize(*pdwRequestedFlags))
+	{
+		if (dwInitialBufferType != 1)
+		{
+			lgd3d_g_bInitialized = 0;
+			DbgReportWarning("Could not Initialize D3D states\n");
+		}
+
+		pcStates->DeInstance();
+		pcRenderBuffer->DeInstance();
+
+		DeleteStatesStack();
+
+		// bit flag?
+		DWORD v4 = *pdwRequestedFlags;
+		BYTE1(v4) = BYTE1(*pdwRequestedFlags) & 0xFE;
+		*pdwRequestedFlags = v4;
+	
+		pcRenderBuffer = cImBuffer::Instance();
+		pcStates = cImStates::Instance();
+
+		DWORD v5 = pcStates->GetRenderStatesSize();
+		CreateStatesStack(10, v5);
```

Comment:
Если имя не определилось, то скорее всего это была временная переменная
```suggestion
		constexpr auto initialSize = 10;
		CreateStatesStack(initialSize, pcStates->GetRenderStatesSize());
```